lib/awful/tooltip.lua

```diff
@@ -546,6 +551,22 @@ function tooltip:set_margins_topbottom(val)
     self:set_margin_topbottom(val)
 end
 
+--- Set the margins between the tooltip and its parent
+--
+-- @DOC_awful_tooltip_gaps_EXAMPLE@
+--
+-- @property tooltip_gaps
+-- @tparam number|table New margins value
+-- @propemits true false
+
+function tooltip:set_gaps(val)
+    self._private.tooltip_gaps = val
```

Comment:
This will need a little more boilerplate code here:

Emit a signal:
```lua
self:emit_signal("property::gaps", val)
```

And change the geometry dynamically:

```lua
if self._private.visible then
    set_geometry(self)
end
```

This way code built on top of the tooltip API will be more reactive/dynamic.

Reply:
Thanks! I realized that the best thing to do after creating all those setups is to contribute to the project itself.